### DISCUSSION TITLE
What are the letters alongside the completion sequences?

### DISCUSSION THREAD

**Discussion by raghavmallampalli** [neoclide/coc.nvim#2938] [Q&A] [answered]
Forgive me if this is an elementary question, but I've been curious about this for a while and I couldn't find anything in the interface section of the help doc.
![image](https://user-images.githubusercontent.com/49435792/109764266-59f22480-7c19-11eb-9b25-205eb4bf059c.png)
In the above screenshot what is the purpose of the letters v, f and [TSC]? I thought they might be information on which language server is providing the contraction. But that doesn't seem to make much sense for some languages ([LS] for python, [LS] or [A] for vim, [TSC] for JS).

**Comment by fannheyward** [comment, marked as answer]
> what is the purpose of the letters v, f and [TSC]

v - variable
f - method or function
[TSC] - Typescript's tsc, tsserver

**Reply by raghavmallampalli** [reply]
Ahh I see. Starting to make sense now. Could you maybe tell me why the server contraction for Python and vim could be LS?

Also is there a full list of these contractions (especially the keyword type contraction) put up somewhere in documentation that I missed?

**Reply by fannheyward** [reply]
[LS] is short for language server.

**Comment by chemzqm** [comment]
Use `:CocList sources` to check sources.
